gnome/gtk-vnc/Portfile

```diff
@@ -33,31 +33,57 @@ checksums           rmd160  ea70671da05ef0024e93858c1ee41387470a4243 \
                     sha256  512763ac4e0559d0158b6682ca5dd1a3bd633f082f5e4349d7158e6b5f80f1ce \
                     size    222204
 
+# Disable unexpected download of subprojects
+meson.wrap_mode     nodownload
+
+set py_ver          3.12
+set py_ver_nodot    [string map {. {}} ${py_ver}]
+
 depends_build-append \
-                    port:pkgconfig \
+                    port:gettext \
+                    path:lib/pkgconfig/gobject-introspection-1.0.pc:gobject-introspection \
                     port:intltool \
+                    path:bin/pkg-config:pkgconfig \
+                    port:python${py_ver_nodot} \
                     path:bin/vala:vala
 
-depends_lib-append  path:lib/pkgconfig/gtk+-3.0.pc:gtk3 \
+depends_lib-append  path:lib/pkgconfig/cairo.pc:cairo \
+                    path:lib/pkgconfig/gdk-pixbuf-2.0.pc:gdk-pixbuf2 \
+                    port:gettext-runtime \
+                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
+                    path:lib/pkgconfig/gtk+-3.0.pc:gtk3 \
                     path:lib/pkgconfig/gnutls.pc:gnutls \
                     port:libgcrypt \
                     port:pulseaudio
 
-patchfiles          MAP_ANONYMOUS.patch
+configure.python    ${prefix}/bin/python${py_ver}
+
+patchfiles-append   patch-meson-python3.diff
+patchfiles-append   MAP_ANONYMOUS.patch
+
+post-patch {
+    reinplace "s|@@PYTHON3_BIN@@|${configure.python}|" \
+                    src/meson.build
+}
 
 if {${os.platform} eq "darwin" && ${os.major} <= 13} {
     # Mavericks comes with Python 2.7.5 which is too old to run keymap-gen properly
     # See https://bugzilla.gnome.org/show_bug.cgi?id=792075
-    depends_build-append port:python27
-    configure.python    ${prefix}/bin/python2.7
+    depends_build-replace \
+                     port:python${py_ver_nodot} \
+                     port:python27
+    configure.python ${prefix}/bin/python2.7
```

Comment:
Why downgrade macOS 10.9 and earlier to python 2.7?

Reply:
> Why downgrade macOS 10.9 and earlier to python 2.7?

I don't have access to my suite of older macOS VMs at the moment, preventing testing on 10.9 and earlier. And since this is the present behavior, I was reluctant to change it.

But happy to eliminate this, if it's not needed?

@danchr Dan, any thoughts on what to do here?

Reply:
> Why downgrade macOS 10.9 and earlier to python 2.7?

Pondering this a bit more, you're totally right; updated to use Python 3.12 across-the-board.